Source/WebKit/WebProcess/WebCoreSupport/WebPermissionController.cpp

```diff
@@ -44,14 +45,17 @@ Ref<WebPermissionController> WebPermissionController::create()
 
 WebPermissionController::WebPermissionController() = default;
 
-void WebPermissionController::query(WebCore::ClientOrigin&& origin, WebCore::PermissionDescriptor&& descriptor, WebCore::Page& page, CompletionHandler<void(std::optional<WebCore::PermissionState>)>&& completionHandler)
+void WebPermissionController::query(WebCore::ClientOrigin&& origin, WebCore::PermissionDescriptor&& descriptor, WebCore::Page* page, WebCore::PermissionQuerySource source, CompletionHandler<void(std::optional<WebCore::PermissionState>)>&& completionHandler)
 {
     auto cachedResult = queryCache(origin, descriptor);
     if (cachedResult != WebCore::PermissionState::Prompt)
         return completionHandler(cachedResult);
 
-    auto pageIdentifier = WebPage::fromCorePage(page).identifier();
-    m_requests.append(PermissionRequest { WTFMove(origin), WTFMove(descriptor), pageIdentifier, WTFMove(completionHandler) });
+    std::optional<WebPageProxyIdentifier> proxyIdentifier = std::nullopt;
+    if (page && (source == WebCore::PermissionQuerySource::Window || source == WebCore::PermissionQuerySource::DedicatedWorker))
+        proxyIdentifier = std::make_optional(WebPage::fromCorePage(*page).webPageProxyIdentifier());
```

Comment:
You probably don't need the `std::make_optional()`

Reply:
Fixed!